src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,34 @@
+package seedu.addressbook.data.person;
+
+import java.util.Objects;
+
+import seedu.addressbook.data.tag.UniqueTagList;
+
+public class Contact {
+	
+	public final String value;
+    private boolean isPrivate;
+
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Phone // instanceof handles nulls
```

Comment:
Make sure that since you pushed the equals method to the Contact class, you should modify all references to the subclasses (e.g. Phone, Email, Address) to refer to the parent class (Contact).